operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/managed/DependentResourceConfigurator.java

```diff
@@ -1,5 +1,9 @@
 package io.javaoperatorsdk.operator.api.reconciler.dependent.managed;
 
+import java.util.Optional;
+
 public interface DependentResourceConfigurator<C> {
```

Comment:
Does this make sense to have, or only the `AnnotationDependentResourceConfiguration`?

Reply:
This is kept for backwards-compatibility… though I guess we're breaking it with the other method…

Reply:
Yeah I guess. So basically the `DependentResourceConfigurator` could be deleted I guess and just have the Annotation version of it?

Reply:
Well, I'm not sure. That's what I thought originally but then people who don't want to use annotations will be forced to implement an interface that won't make sense to them…  Maybe we still need to clean some things up here. I'd be in favor of doing this in subsequent PRs, though.